### PR TITLE
Pseudomentatory parameters in ES6 functions 

### PR DESCRIPTION
**Pull request by AvraamMavridis**

### CODE REVIEW

**Comment by parro-it** [comment]
nice tip!

**Comment by afc163** [comment]
:+1:

**Comment by AvraamMavridis** [comment]
thx!

**Comment by loverajoel** [comment]
@AvraamMavridis can you add a link to official documentation about es6?

**Comment by AvraamMavridis** [comment]
@loverajoel you mean about default parameters? sure.

File: README.md [hunk 1]
```diff
@@ -18,6 +18,26 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
 
 # Tips list
 
+## #10 - Pseudomentatory parameters in ES6 functions
```
Comment (loverajoel) [inline README.md]: update number to 11

File: README.md [hunk 2]
```diff
@@ -18,6 +18,26 @@ To get updates, watch the repo and follow the [Twitter account](https://twitter
 
 # Tips list
 
+## #10 - Pseudomentatory parameters in ES6 functions 
+
+> 2016-01-10 by [Avraam Mavridis](https://github.com/AvraamMavridis)
```
Comment (loverajoel) [inline README.md]: update date to today

**Comment by loverajoel** [comment]
@AvraamMavridis yes please, after your changes I will merge and publish this tip :rocket:

**Comment by knpwrs** [comment]
Wow. That's really cool. It's a side effect of the default parameter expression not being evaluated until the value is actually required. Once we get good support for [proxies](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) we could have a thrower object that automatically throws any property accessed on it.

``` js
const required = new Proxy({},  {
  get: function(target, name) {
    throw new Error(`${name} is a required parameter.`);
  }
});

const getSum = (a = required.a, b = required.b) => a + b;
```

I believe that should work. I don't actually have any environments right now that can run proxies.

**Comment by AvraamMavridis** [comment]
@loverajoel done :boom:

**Comment by loverajoel** [comment]
@davecan I can't merge you have to do a mergeback or rebase. Will be better if you can squash your commits

**Comment by AvraamMavridis** [comment]
@loverajoel I guess wrong @? Commits squashed.

**Comment by loverajoel** [comment]
@AvraamMavridis you did wrong the mergeback -> https://github.com/loverajoel/jstips/pull/60/files

**Comment by loverajoel** [comment]
@AvraamMavridis thanks for your contribution!

**Comment by AvraamMavridis** [comment]
@loverajoel thx!

**Comment by AvraamMavridis** [comment]
@knpwrs sweet. thx for that.

**Pull request merged**